File: Infinite Scroll (ObjC)/InfiniteScrollSample/ArticleCell.m

#import "ArticleCell.h"
#import <QuartzCore/QuartzCore.h>


@interface ArticleCell()
@end

@implementation ArticleCell

- (void)prepareForReuse {
    self.adImageView.image = nil;
    [super prepareForReuse];
}

- (void)displaySponsoredIndicators:(BOOL)isSponsored {
    if (isSponsored) {
        [self.sponsoredLabel setHidden:NO];
        [self.sponsoredIndicator setHidden:NO];
        [self.contentView setBackgroundColor:[UIColor colorWithRed:230/255.0 green:235/255.0 blue:242/255.0 alpha:1]];
    } else {
        [self.sponsoredLabel setHidden:YES];
        [self.sponsoredIndicator setHidden:YES];
        [self.contentView setBackgroundColor:[UIColor whiteColor]];
    }
}

@end

File: Table View (Swift)/NativoSwiftTableViewSample/ArticleCell.swift
//
//  ArticleCell.swift
//  NativoSwiftTableViewSample
//
//  Copyright © 2019 Nativo. All rights reserved.
//

import UIKit
import NativoSDK

class ArticleCell: UITableViewCell, NtvAdInterface {
    
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var authorNameLabel: UILabel!
    @IBOutlet weak var dateLabel: UILabel!
    @IBOutlet weak var previewTextLabel: UILabel!
    @IBOutlet weak var adImageView: UIImageView!
    @IBOutlet weak var sponsoredLabel: UILabel!
    @IBOutlet weak var sponsoredIndicator: UIImageView!
    
    
    func displaySponsoredIndicators(_ isSponsored: Bool) {
        if isSponsored {
            self.sponsoredLabel.isHidden = false
            self.sponsoredIndicator.isHidden = false
            self.contentView.backgroundColor = UIColor.init(red: 230/255.0, green: 235/255.0, blue: 242/255.0, alpha: 1)
        } else {
            self.sponsoredLabel.isHidden = true
            self.sponsoredIndicator.isHidden = true
            self.contentView.backgroundColor = UIColor.white
        }
    }
 
    func shouldPrependAuthorByline() -> Bool {
        return false
    }
    

    override func awakeFromNib() {
        super.awakeFromNib()
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
    }

}

